File: PersonalBlog.API/Models/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace PersonalBlog.API.Models
{
    public class Post
    {
        public int Id { get; set; }
        [Required]
        [StringLength(150,MinimumLength = 3)]
        public string Title { get; set; }
        public string Slug { get; set; }
        [Required]
        [StringLength(600 , MinimumLength = 3)]
        public string Text { get; set; }
        [Required]
        public string Author { get; set; }    
        public bool IsPublished { get; set; }
        [DataType(DataType.Date)]
        public DateTime PubDate  { get; set; }
        public bool CommentOpen { get; set; }
        public string ArticleSource { get; set; }
        public ICollection<Tag> Tags { get; set; }
        public ICollection<Comment> Comments { get; set; }
        public ICollection<Attachment> Attachments { get; set; }
    }
}

File: PersonalBlogSPA/src/app/_models/post.ts
import { Attachment } from './attachment';
import { Tag } from './tag';
import { Comment } from './comment';


export interface Post {
    id: number;
    title: string;
    slug: string;
    text: string;
    author: string;
    pubDate?: Date;
    commentOpen?: boolean;
    articleSource?: string;
    tags?: Tag[];
    comments?: Comment[];
    attachment?: Attachment[];

}
